cleverhans/utils.py

```diff
@@ -220,6 +220,95 @@ def grid_visual(data):
     return figure
 
 
+def get_logits_over_interval(sess, model, x, adv_x, x_data,
+                             min_epsilon=-10, max_epsilon=10,
+                             num_points=21):
+    """Get logits when the input is perturbed in an interval in adv direction.
+
+    Args:
+        sess: Tf session
+        model: Model for which we wish to get logits
+        x: Tf placeholder for data
+        adv_x_real: Tf placeholder for adv perturbation of x
+        x_data: Numpy array corresponding to the data slice used to generate plot
+        min_epsilon: Minimum value of epsilon over the interval
+        max_epsilon: Maximum value of epsilon over the interval
+        num_points: Number of points used to interpolate
+
+    Returns:
+        Numpy array containing log probabilities
+
+    Raises:
+        ValueError if min_epsilon is larger than max_epsilon
+    """
+    if min_epsilon > max_epsilon:
+        raise ValueError('Minimum epsilon is less than maximum epsilon')
+
+    delta = (max_epsilon - min_epsilon) / (num_points - 1)
+    epsilon = min_epsilon
+    eta = adv_x - x
+    for i, epsilon in enumerate(np.arange(min_epsilon, max_epsilon + 1, step=delta)):
```

Comment:
why add 1 to max_epsilon?

Reply:
because it wouldn't take the max_epsilon value otherwise, I switched to using np.linspace which is recommended for floats.